Extras, user and release informations not showing up when using Native SDK

## Important Details

How are you running Sentry?

* [ ] On-Premise docker [Version xyz]
* [x] Saas (sentry.io)
* [ ] Other [briefly describe your environment]

## Description

We were successful in building the SDK for macOS with the crashpad integration and we have successfully configured it so when it crashes we do see the event in the dashboard.

Crash is generated as followed:
```cpp
int* a = nullptr;
*a = 1;
```
However, it seems that we cannot get some information to show on the "event" such as the user, the release and some extras.

Here is how we use it :

```cpp
sentry_value_t user = sentry_value_new_object();
sentry_value_set_by_key(user, "id", sentry_value_new_int32(123456789));
sentry_value_set_by_key(
    user, "username", sentry_value_new_string("John Doe"));
sentry_set_user(user);
sentry_set_extra(
    "bitrate", sentry_value_new_int32(g_StartupOption.bitrate));
sentry_options_set_release(options, "1.0.0");
```

However, we did manage to add attachments which are working well.
I hope you'll be able to enlighten us on what we are doing wrong or what we misunderstood.

### What you expected to happen

We should see the release in the event as well as user info and all extras.

I have tried to build `master` but i get this when trying to make `sentry_crashpad`:
```
==== Building crashpad_util (release) ====
==== Building crashpad_minichromium_base (release) ====
==== Building crashpad_client (release) ====
==== Building crashpad_snapshot (release) ====
==== Building crashpad_minidump (release) ====
make[1]: *** No rule to make target `../crashpad/build/crashpad/client/annotation.cc', needed by `obj/Release/crashpad_client/annotation.o'.  Stop.
make: *** [crashpad_client] Error 2
make: *** Waiting for unfinished jobs....
make[1]: *** No rule to make target `../crashpad/build/crashpad/third_party/mini_chromium/mini_chromium/base/debug/alias.cc', needed by `obj/Release/crashpad_minichromium_base/alias.o'.  Stop.
make: *** [crashpad_minichromium_base] Error 2
make[1]: *** No rule to make target `../crashpad/build/crashpad/minidump/minidump_annotation_writer.cc', needed by `obj/Release/crashpad_minidump/minidump_annotation_writer.o'.  Stop.
make: *** [crashpad_minidump] Error 2
make[1]: *** No rule to make target `../crashpad/build/crashpad/snapshot/annotation_snapshot.cc', needed by `obj/Release/crashpad_snapshot/annotation_snapshot.o'.  Stop.
make: *** [crashpad_snapshot] Error 2
make[1]: *** No rule to make target `../crashpad/build/crashpad/util/mach/child_port.defs', needed by `gen/util/mach/child_portUser.c'.  Stop.
make[1]: *** Waiting for unfinished jobs....
Creating obj/Release/crashpad_util
```

also here is what i see in Sentry just so you understand what I expect to see
<img width="847" alt="Capture d’écran 2019-11-14 à 11 03 18" src="https://user-images.githubusercontent.com/8132834/68847172-830e3f00-06ce-11ea-9d43-f2c7cebb3c0c.png">

You can see that it does not mention any release and the user seems empty.
When scrolling down I do not see any extra.

Also, when sneak peeking the JSON file, I do not find anything about the user or other things. Iwas expecting to see "John Doe" somewhere.

I  am having issues on macOS but it seems to be the same on all our platforms (Windows and Linux).

I cannot get `master` to build unfortunatly: 
```
==== Building crashpad_util (release) ====
==== Building crashpad_minichromium_base (release) ====
==== Building crashpad_client (release) ====
==== Building crashpad_snapshot (release) ====
==== Building crashpad_minidump (release) ====
make[1]: *** No rule to make target `../crashpad/build/crashpad/client/annotation.cc', needed by `obj/Release/crashpad_client/annotation.o'.  Stop.
make: *** [crashpad_client] Error 2
make: *** Waiting for unfinished jobs....
make[1]: *** No rule to make target `../crashpad/build/crashpad/third_party/mini_chromium/mini_chromium/base/debug/alias.cc', needed by `obj/Release/crashpad_minichromium_base/alias.o'.  Stop.
make: *** [crashpad_minichromium_base] Error 2
make[1]: *** No rule to make target `../crashpad/build/crashpad/minidump/minidump_annotation_writer.cc', needed by `obj/Release/crashpad_minidump/minidump_annotation_writer.o'.  Stop.
make: *** [crashpad_minidump] Error 2
make[1]: *** No rule to make target `../crashpad/build/crashpad/snapshot/annotation_snapshot.cc', needed by `obj/Release/crashpad_snapshot/annotation_snapshot.o'.  Stop.
make: *** [crashpad_snapshot] Error 2
make[1]: *** No rule to make target `../crashpad/build/crashpad/util/mach/child_port.defs', needed by `gen/util/mach/child_portUser.c'.  Stop.
make[1]: *** Waiting for unfinished jobs....
Creating obj/Release/crashpad_util
```

--- a/src/api.cpp
+++ b/src/api.cpp
@@ -38,6 +38,10 @@ int sentry_init(sentry_options_t *options) {
     }
     cleanup_old_runs();
 
+    // make sure that the scopes are at least flushed once after the backend
+    // is started for the initial data to be written.
+    Scope::with_scope_mut([](Scope &_scope) { /* run for side effect */ });
+
     if (g_options->transport) {
         g_options->transport->start();
     }

